db/migrations/postgres/000083_threads_threaddeleteat.up.sql

```diff
@@ -0,0 +1,5 @@
+-- Drop any existing DeleteAt column from 000081_threads_deleteat.up.sql
+ALTER TABLE threads DROP COLUMN IF EXISTS deleteat;
+
+ALTER TABLE threads ADD COLUMN IF NOT EXISTS threaddeleteat bigint;
+UPDATE threads SET threaddeleteat = posts.deleteat FROM posts WHERE threads.threaddeleteat IS NULL AND posts.id = threads.postid;
```

Comment:
Can you check the query plan for this? Just to be sure if postgres check the `IS NULL` condition first. The mysql and postgres optimization may differ.

Reply:
Postgres:

```
explain UPDATE threads SET threaddeleteat = posts.deleteat FROM posts WHERE threads.threaddeleteat IS NULL AND posts.id = threads.postid;
                                     QUERY PLAN
-------------------------------------------------------------------------------------
 Update on threads  (cost=0.56..73380.96 rows=1 width=147)
   ->  Nested Loop  (cost=0.56..73380.96 rows=1 width=147)
         ->  Seq Scan on threads  (cost=0.00..73378.18 rows=1 width=121)
               Filter: (threaddeleteat IS NULL)
         ->  Index Scan using posts_pkey on posts  (cost=0.56..2.78 rows=1 width=41)
               Index Cond: ((id)::text = (threads.postid)::text)
(6 rows)
```

MySQL:
```
mysql> EXPLAIN UPDATE Threads, Posts
    -> SET Threads.ThreadDeleteAt = Posts.DeleteAt
    -> WHERE Posts.Id = Threads.PostId
    -> AND Threads.ThreadDeleteAt IS NULL;
+----+-------------+---------+------------+--------+---------------+---------+---------+-----------------------------+---------+----------+-------------+
| id | select_type | table   | partitions | type   | possible_keys | key     | key_len | ref                         | rows    | filtered | Extra       |
+----+-------------+---------+------------+--------+---------------+---------+---------+-----------------------------+---------+----------+-------------+
|  1 | UPDATE      | Threads | NULL       | ALL    | PRIMARY       | NULL    | NULL    | NULL                        | 1505485 |    10.00 | Using where |
|  1 | SIMPLE      | Posts   | NULL       | eq_ref | PRIMARY       | PRIMARY | 106     | jswithtafmdb.Threads.PostId |       1 |   100.00 | NULL        |
+----+-------------+---------+------------+--------+---------------+---------+---------+-----------------------------+---------+----------+-------------+
2 rows in set (0.00 sec)
```